Rename packages 2

monkey3310 opened the pull request:
<!-- Thanks for submitting a pull request! Please provide enough information so that others can review your pull request. The two fields below are mandatory. -->

<!-- Please remember to update CHANGELOG.md in the root of the project if you have not done so. -->

## Summary

This is part of #4969 - standardizes file names in `packages/jest-worker` and `packages/pretty-format` under Facebook naming conventions

## Test plan

<!-- Demonstrate the code is solid. Example: The exact commands you ran and their output, screenshots / videos if the pull request changes UI. -->

I've run yarn test and although its not passing, I'm getting same test failures on master

thymikee commented on CHANGELOG.md:
```diff
@@ -95,6 +95,8 @@
 - `[docs]` Add correct default value for `testUrl` config option ([#7277](https://github.com/facebook/jest/pull/7277))
 - `[jest-util]` [**BREAKING**] Remove long-deprecated globals for fake timers ([#7285](https://github.com/facebook/jest/pull/7285))
 - `[docs]` Remove duplicate code in `MockFunctions` ([#7297](https://github.com/facebook/jest/pull/7297))
+- `[jest-haste-map]` Standardize filenames ([#7316](https://github.com/facebook/jest/pull/7316))
```
jest-haste-map -> jest-worker

thymikee commented on packages/jest-worker/src/__tests__/worker.test.js:
```diff
@@ -38,7 +38,7 @@ beforeEach(() => {
     return forkInterface;
   });
 
-  Worker = require('../worker').default;
```
can we rename the test file to correspond to new `Worker.js`?

thymikee commented on packages/pretty-format/src/plugins/lib/markup.js:
```diff
@@ -9,7 +9,7 @@
 import type {Config, Printer, Refs} from 'types/PrettyFormat';
 
-import escapeHTML from './escape_html';
+import escapeHTML from './escapeHTML';
```
Can you rename the rest of the files in pretty-format? There's quite a bunch of left

monkey3310 replied:
Hey, I might have missed some, could you point out what files are left? I thought that the ones in `/src/` have multiple exports per file, hence I left files with snake_case

thymikee replied:
We want to rename _every_ file that doesn't stick to the convention.
E.g. 
```
packages/pretty-format/src/__tests__/asymmetric_matcher.test.js
packages/pretty-format/src/plugins/asymmetric_matcher.js
```
to have names as variables:
```js
import AsymmetricMatcher from './plugins/asymmetric_matcher';
import ConvertAnsi from './plugins/convert_ansi';
import DOMCollection from './plugins/dom_collection';
import DOMElement from './plugins/dom_element';
import Immutable from './plugins/immutable';
import ReactElement from './plugins/react_element';
import ReactTestComponent from './plugins/react_test_component';
```

monkey3310 replied:
But the `'./plugins/asymmetric_matcher'` has 3 exports, so according to those:

> All folders -> kabab-case
All files that export a single Class -> CapitalizeCase
All files that export a single function -> camelCase
All other files -> underscore_case

its falling into the `underscore_case`?

thymikee replied:
This is a mistake in interface, they shouldn't expose `serialize` and `test` directly, as they do it in a default export – and we only use it that way. You can remove those exports. cc @pedrottimark

monkey3310 replied:
@thymikee Great, I will redo the exports and fix those files, thanks 👍

monkey3310 commented:
@thymikee Hey, I've renamed plugins, but now I got quite a lot of test failures, related to the fact that I removed the exports for `test` / `serialize` from them, and seems like they are still accessed directly: `plugins[p].test is not a function`. Any tip on how should I proceed with this?

thymikee commented:
Oh, that's interesting. For now let's just bring back the exports if they're just for testing

monkey3310 commented:
@thymikee 👍 reverted the mentioned exports and now everything is passing

thymikee commented on packages/pretty-format/src/__tests__/Immutable.test.js:
```diff
@@ -11,7 +11,7 @@
 
 import React from 'react';
 import Immutable from 'immutable';
-import {getPrettyPrint} from './expect_util';
```
The `getPrettyPrint` now only exports a single fn with the same name. Since there's no other exports there, maybe it's worth to change it to default export?

monkey3310 replied:
Did the changes, not sure if that's what you meant, please take a look 👍
